### PR TITLE
Close #687, pass sources path in through the command line

### LINKED ISSUE #687: Pass feature/sources tests path through the command line
This is fine for now, but I didn't see an issue anywhere, so I wanted to note; I think the next refactor here should be do add a proper command line argument handler like [minimist](https://github.com/minimistjs/minimist), and then pass these two paths through the command line instead. It has a few benefits:
* ~removes the necessity of 2 additional environment variables as well, `get_user_id`, and `get_user_project_name`~
* puts docassemble specific knowledge (what these specific paths are) in the docassemble specific package (ALKilnInThePlayground)
* makes it more flexible in general; I made a quick little tool that could run 10 different versions of Kiln at the same time by running them in different working directories and pointing them to the source directory elsewhere. 

I did a little preliminary work on this in https://github.com/SuffolkLITLab/ALKiln/commit/f8e9921788abc3c45748a8090e451275081145e9, but I'm trying to shift priorities for a bit, and don't want this to be slowed down by me.

_Originally posted by @BryceStevenWilley in https://github.com/SuffolkLITLab/ALKiln/pull/686#discussion_r1160031854_

In response: Great points!

Edit:
~We may add to this the base url of the interview. The urls are different depending on whether the package was installed in a server's Playground, or on the server itself. This base url might also be passed into the command line instead of `install_type` config/env var. See comments/discussions about `install_type` in https://github.com/SuffolkLITLab/ALKiln/pull/753 and https://github.com/SuffolkLITLab/ALKiln/pull/740.~

Also, if we don't need `get_user_id` for checking whether the server is up, we might use `server_url.com/health_check?ready=1`. Still have to check the validity of the API key somehow.

### PR DESCRIPTION
**Pull request by plocket**
~Edit: Blocked by https://github.com/SuffolkLITLab/ALKiln/issues/760.~

~Will have to update~ Have updated ALKilnInThePlayground.

Is this format what we want?

``` bash
npm run test @tag --sources=path/to/sources/dir
```

In this PR, I have:

* ~Added tests for any new features or bug fixes (if relevant)~
* [x] Added my changes to the CHANGELOG.md at the top, under the "Unreleased" section
* [x] Ensured issues that this PR closes will be [automatically closed](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

### Reason for this PR

This allows the caller to specify the path to the sources folder, something that's very environment specific and hard for ALKiln itself to work out. See the issue.

### Links to any solved or related issues

Closes #687 

### Any manual testing I have done to ensure my PR is working

Tested locally with:

```
npm run cucumber @none -- --sources=./none
npm run cucumber @none -- --sources=./none --sources=./docassemble/*/data/sources
npm run cucumber @none -- --sources=./docassemble/*/data/sources
```

Tested from ALKilnInThe Playground as well.

### CODE REVIEW

**Comment by plocket** [comment]
Conclusion: That is indeed the desired format for the command line args.

**Comment by plocket** [comment]
This will have some merge conflicts with #774. I'd like to prioritize that one, so I'm holding off until that one's merged.

**Comment by plocket** [comment]
I think this may be unblocked now, and ready for review.

**Comment by plocket** [comment]
I think this may be unblocked now, and ready for review. If you have bandwidth, @BryceStevenWilley, would love your input.

File: package.json
Comment (BryceStevenWilley) [inline package.json:3]: Should probably change this:

```suggestion
  "version": "5.4.1-sources-8",
```

File: lib/run_cucumber.js
Comment (BryceStevenWilley) [inline lib/run_cucumber.js]: Given that we're using `fg.sync( {suppressErrors:True...})`, I think including all of these as defaults wouldn't hurt.

File: lib/scope.js
Comment (BryceStevenWilley) [inline lib/scope.js]: Not sure I get this change. Why would you remove this extra information? Without context, the former report msg is clearer IMO.

**Review by BryceStevenWilley — COMMENTED** [review]
LGTM, tiny nit comments

File: package.json
Comment (plocket) [reply package.json:3]: I'm never sure how to handle this. `v5.1.0-sources-7` is the one that has been published. `v5.4.1-sources-8` doesn't exist, though on a surface level it more accurately describes where in commit history this now sits. I'll be publishing it under a new version anyway, so I think I'm going to keep what I have and resolve this.

File: lib/scope.js
Comment (plocket) [reply lib/scope.js]: To get into this nit, if you're interested, this is just my gut feeling: I feel it muddies the waters because it doesn't necessarily line up with the step they wrote. Getting inaccurately specific can sometimes be more confusing than being more general. They either gave file(s) to "upload" or they gave file(s) to "compare". They didn't mention "finding" anything and they don't know what the code is doing in the background. Again, just my gut feeling. We could try asking a user, though, either now or at some future point.

File: lib/run_cucumber.js
Comment (plocket) [inline lib/run_cucumber.js]: Do you mean like this?

```suggestion

  // We'll use the user info to construct the possible paths to /sources
  let id = session_vars.get_user_id();
  // `project` is the name in the server docassemble "Project" page
  let project = session_vars.get_user_project_name();

  // process.argv is a list of strings of commands and args.
  // If using `npm run cucumber`, use `--` before `--sources=./foo`
  let argv = require(`minimist`)( process.argv.slice(2), {
    default: {
      sources: [
        `./docassemble/*/data/sources`,  // In GitHub folder
        `/usr/share/docassemble/files/playgroundsources/${id}/${project}/*.feature`, // From playground without S3
        `/tmp/playgroundsources/${id}/${project}/*.feature` // From playground with S3
      ],
    }
  });
```

File: lib/run_cucumber.js
Comment (plocket) [reply lib/run_cucumber.js]: I made an example lower down of what I _think_ you mean

File: lib/scope.js
Comment (BryceStevenWilley) [reply lib/scope.js]: I get that, but I don't think this make it any clearer to the user. If I was a user, the first thing I'd think if I saw this error message, I'd think "what list?". If they only intended to pass one file to upload (even if they didn't pass any)? In their mind, they didn't pass a list, so this message is still confusing. It should be followed by a specific error message from the upload or compare steps anyway.

I guess this gets into who the messages are for. When they contain relatively little information like this, IMO they should at least be useful to developers. Several months out, the former message would be still be useful to me, whereas the latter wouldn't, and depending on how the user is communicating their issue with them, it either adds a simple search, or a lot more back and forth than necessary.

File: lib/run_cucumber.js
Comment (BryceStevenWilley) [reply lib/run_cucumber.js]: Yeah! I'd make sure the tests run fine, we can't really test this because it's not gonna be used in ALKilnInThePlayground, but it can't hurt, I like it a lot better than commented code.

File: lib/scope.js
Comment (plocket) [reply lib/scope.js]: Good point. What do you think of `There are no file(s) to find written in your Step.`? I'll push that for now, dependent on your input. [Or maybe I'll try to think of something with less passive voice.]

Comment (plocket) [reply lib/scope.js]: `Your Step is missing the names of which file(s) to find.`

Comment (BryceStevenWilley) [reply lib/scope.js]: I like the `Your step is missing...` one!

**Review by BryceStevenWilley — APPROVED** [review]

**Pull request merged**
